docs/tutorials/05-external-jsonschema.md

```diff
@@ -0,0 +1,311 @@
+```
+
+```
+component-cli ca remote push ./docs/tutorials/resources/external-jsonschema/definitions
+```
+
+</details>
+
+#### Echo Server Deployment
+
+The first component is now created and uploaded to the registry.
+Now the user of the jsonschema is created.
+
+The echo server example is reused and enhanced with another import parameter _resources_.
+
+Before the jsonschema can be used in the blueprint, it has to be referenced in the component descriptor so that Landscaper knows about the dependency towards the `definitions` component.
+Subsequently, a component reference is added (see the complete component descriptor in the details):
+```yaml
+componentReferences:
+- name: definitions
+componentName: github.com/gardener/landscaper/external-jsonschema/definitions
+version: v0.1.0
+```
+
+<details>
+<div id="echo-server-comp-desc"></div>
+
+```yaml
+# ./docs/tutorials/resources/external-jsonschema/echo-server/component-descriptor.yaml
+meta:
+  schemaVersion: v2
+
+component:
+name: github.com/gardener/landscaper/external-jsonschema/echo-server
+version: v0.1.0
+
+provider: internal
+
+repositoryContexts:
+- type: ociRegistry
+  baseUrl: eu.gcr.io/gardener-project/landscaper/tutorials/components
+
+sources: []
+componentReferences:
+- name: definitions
+  componentName: github.com/gardener/landscaper/external-jsonschema/definitions
+  version: v0.1.0
+
+resources:
+- type: ociImage
+  name: echo-server-image
+  version: v0.2.3
+  relation: external
+  access:
+  type: ociRegistry
+  imageReference: hashicorp/http-echo:0.2.3
+```
+
+</details>
+
+When the component reference is added to the echo-server's component descriptor, the resource import can be added to the blueprint.
+The import also contains a valid jsonschema but uses the `$ref` attribute with a custom landscaper protocol implementation.
+That custom protocol specifies that the jsonschema should be fetched from the resource `resources-definition` of the referenced component `definitions`.
+
+For detailed explanation about available reference methods see the [blueprint jsonschema documentation](../usage/JSONSchema.md)
+
+```yaml
+imports:
+- name: resources
+  schema:
+    $ref: "cd://componentReferences/definitions/resources/resources-definition"
+```
+
+The blueprint is defined, so it can be added to the component descriptor using the component-cli and then uploaded.
+See the details section for concrete resource definitions and cli calls.
+
+<details>
+
+```yaml
+# ./docs/tutorials/resources/external-jsonschema/echo-server/blueprint.yaml
+apiVersion: landscaper.gardener.cloud/v1alpha1
+kind: Blueprint
+
+imports:
+- name: cluster
+  targetType: landscaper.gardener.cloud/kubernetes-cluster
+- name: ingressClass
+  schema:
+    type: string
+- name: resources
+  schema:
+    $ref: "cd://componentReferences/definitions/resources/resources-definition"
+
+deployExecutions:
+- name: default
+  type: GoTemplate
+  file: /defaultDeployExecution.yaml
```

Comment:
should the DeployExecution which consumes the resources import be listed here too?

Reply:
oh yes good point